simulation.c

```diff
@@ -419,6 +421,101 @@ void stepSimulation(void)
    }
 }
 
+void stepGamemode(int timeremain)
+{
+   int ip;
+   //tick approx once per second:
+   if( !(timeremain % 64) )
+   {
+      if(conf.gametype != CONFIG_GAMETYPE_DEFAULT)
+      {
+         switch(conf.gametype)
+         {
+            case CONFIG_GAMETYPE_PREFERED_TARGET:
+            {
+               static int currentPreferedTarget;
+               char choosenew=1;
+               //Find active player that is prefered target
+               for(ip = 0; ip < conf.maxPlayers; ++ip)
+               {
+                   SimPlayer* p = &(player[ip]);
+                   if(p->active && p->extrapoints)
+                   {
+                      choosenew=0;
+                      break;
+                   }
+               }
+               
+               if(choosenew)
+               {
+                  //no prefered target. Choose a new one:
+                  ip=(currentPreferedTarget+1) % conf.maxPlayers;
+                  for(ip=(currentPreferedTarget+1) % conf.maxPlayers;
+                      ip != currentPreferedTarget; ip=(ip+1)% conf.maxPlayers)
+                  {
+                      SimPlayer* p = &(player[ip]);
+                      if(p->active)
+                      {
+                         printf("%d is new prefered target\n",ip);
+                         p->extrapoints=1;
+                         currentPreferedTarget=ip;
+                         break;
+                      }
+                  }
+               }
+               break;
+            }
+            case CONFIG_GAMETYPE_KILL_OLDEST:
+            {
+               static int tickcount;
+               ++tickcount;
+               if(!(tickcount%64))
+               {
+                  for(ip = 0; ip < conf.maxPlayers; ++ip)
+                  {
+                      SimPlayer* p = &(player[ip]);
+                      if(p->active && p->extrapoints<3)
+                      {
+                         ++p->extrapoints;
+                         break;
+                      }
+                  }
+                  tickcount=0;
+               }
+               break;
+            }
+            case CONFIG_GAMETYPE_KILL_BEST:
```

Comment:
I like this variant the best, however: Currently, if you have multiple players with the same number of kills, the first one is chosen. Better: choose the one, which has fewer deaths. If they also have the same number of deaths, both should be worth an extrapoint. (See also the sorting in display.c for end of round....)